### PR TITLE
Kubernetes hostname changes

### PR DESCRIPTION
**Pull request by nobletrout**
fixed my fork, made git close pr #5559 sorry.

### CODE REVIEW

**Comment by alles-klar** [comment]
@dsever I suggest we continue with this PR because it fixes the problem from @nobletrout and doesn't change to much. I agree with you that there is room for improvement, but maybe we continue with it in later PR, when someone need the feature.

**Review by alles-klar — APPROVED** [review]

**Comment by dsever** [comment]
@alles-klar, problem I see here is that PR breaks upgrades. Previously there was def as: `key:value` and now is `key: list`.  

At least it has to be documented.

**Comment by alles-klar** [comment]
@nobletrout I added an upgrade note, but I think we must also update the [kubernetes instructions](https://github.com/DefectDojo/django-DefectDojo/blob/master/readme-docs/KUBERNETES.md). Can you please take a look and test/update it?

**Comment by dsever** [comment]
Guys finally I had time to test it in minikube, sorry about that (my bad). So few comments from my side, it works as expected, that is good. 

But I'm thinking about something else, this change introduces many changes but actually it changes only  `DD_ALLOWED_HOSTS:` it is not touching items like liveness probes, and it break changes (if someone has configured CICD way of working). Addition to that `hosts` is list, and for liveness probes and TLS certs it is taking first element:

`{{- $hostName := index .Values.hosts 0}}` 

So if order is not respected ti will break the deployment. 

Maybe would be better to introduce new value like e.g  `alternative_hosts` that will be for DD_ALLOWED_HOSTS translated to:
`host` + `alternative_hosts` (list) 

So user will not need to break/adapt current installation, not pay attention of the order, but required functionality will be still there.

**Review by dsever — COMMENTED** [review]
Please take a look at my last comment.

**Comment by nobletrout** [comment]
Makes sense I was just trying to avoid adding an additional value.

File: helm/defectdojo/templates/configmap.yaml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 {{- $fullName := include "defectdojo.fullname" . -}}
+{{- $hosts := .Values.host -}}
```
Comment (dsever) [inline helm/defectdojo/templates/configmap.yaml:2]: Can you move it to tpl?

**Comment by dsever** [comment]
Now it looks good to me, I don't think  integration tests that fails has anything to do with your changes.

**Comment by dsever** [comment]
@nobletrout can you try to rebase it.

**Comment by dsever** [comment]
Am I right, you did rebase against master instead of dev?

**Comment by nobletrout** [comment]
ugh, looks like i did. the github UI isn't very clear about how that sort of stuff works.

**Comment by nobletrout** [comment]
git you make everything very hard.

**Review by dsever — APPROVED** [review]
It looks ok now

File: helm/defectdojo/values.yaml [hunk 2]
```diff
@@ -20,7 +20,14 @@ createPostgresqlSecret: false
 # Option to use "postgresql" or "mysql" database type, by default "mysql" is chosen
 # Set the "enable" field to true of the database type you select (if you want to use internal database) and false of the one you don't select
 database: postgresql
+# List of allowed hosts, first entry is used for http header
```
Comment (alles-klar) [inline helm/defectdojo/values.yaml:23]: This comment is not correct anymore. It isn't a list. Can you please update it?

Comment (nobletrout) [reply helm/defectdojo/values.yaml]: done

**Pull request merged**